test/puppetlabs/comidi_test.clj

```diff
@@ -288,3 +288,43 @@
                       routes->handler
                       wrap-with-fake-match-context)]
       (is (= "bar!" (handler {:uri "/bunk"}))))))
+
+(deftest wrap-leaves-with-middleware-test
+  (let [inner-middleware (fn [handler]
+                           (fn [request]
+                             (update-in (handler request) [:body] #(str "inner-" %))))
+        bb-wrapper-middleware (fn [handler]
+                                (fn [request]
+                                  (update-in (handler request) [:body] #(str "bb-wrapper-" %))))
+        outer-middleware (fn [handler]
+                           (fn [request]
+                             (update-in (handler request) [:body] #(str "outer-" %))))
+        aa-route (GET "/aa" request "aa!")
+        bb-route (ANY "/bb" request "bb!")
+        cc-route (ANY "/cc" request "cc!")
+        dd-route (DELETE "/dd" request "dd!")
+        ee-route (GET "/ee" request "ee!")
+        ff-route (ANY "/ff" request "ff!")
+        left-routes (context "/left" aa-route bb-route)
+        middle-routes (context "/middle" cc-route dd-route)
+        right-routes (context "/right" ee-route ff-route)
+        handler (-> (routes left-routes middle-routes right-routes) routes->handler)]
+    (is (= (:body (handler {:uri "/left/aa" :request-method :get})) "aa!"))
+    (is (= (:body (handler {:uri "/left/bb" :request-method :post})) "bb!"))
+    (is (= (:body (handler {:uri "/middle/cc" :request-method :get})) "cc!"))
+    (is (= (:body (handler {:uri "/middle/dd" :request-method :delete})) "dd!"))
+    (is (= (:body (handler {:uri "/right/ee" :request-method :get})) "ee!"))
+    (is (= (:body (handler {:uri "/right/ff" :request-method :delete})) "ff!"))
```

Comment:
I think this test would be a bit clearer if there were two `testing` blocks, one above this block of assertions saying that it's testing the routes before any middleware is applied, and another around the next set of assertions, saying that it's the routes after `wrap-routes` is used.